setuptools_scm/hg.py

```diff
@@ -8,13 +8,18 @@
 def _hg_tagdist_normalize_tagcommit(root, tag, dist, node):
     dirty = node.endswith('+')
     node = node.strip('+')
-    st = do('hg st --no-status --change %s' % str(node), root)
-
-    trace('normalize', locals())
-    if int(dist) == 1 and st == '.hgtags' and not dirty:
-        return meta(tag)
+    revset = ("(branch(.) and tag('{0}')::. and file('re:^(?!\.hgtags).*$')"
+              " - tag('{0}'))").format(str(tag))
```

Comment:
I'm not sure repr is guaranteed to use simple quotes. `repr("abc")` is `'abc'`, but if it would ever be `"abc"` that would break the revset. Whereas with `str` (or `{tag!s}) you get the plain string without quotes.